Mhv 35381 older messages styling

alex-morgun-1 opened the pull request:
## Description
![image](https://user-images.githubusercontent.com/87077843/187755896-25a843cb-10fb-4c9b-ad82-4be46fa0da4d.png)


## Original issue(s)
[JIRA - SM to VA.gov: Fix styling for Older Messages component](https://vajira.max.gov/browse/MHV-35381)


## Testing done


## Screenshots


## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

micahchiang committed:
initial redux code

micahchiang committed:
merge master

micahchiang committed:
inbox view and message card

micahchiang committed:
pagination, sorting

mmoyer-va committed:
Merge branch 'main' into mhv-sm-main

mmoyer-va committed:
MHV-33575 Changed import to fix error 'node_modules\web-components doesn't exist'

mmoyer-va committed:
Merge remote-tracking branch 'origin/main' into mhv-sm-main

mmoyer-va committed:
[#MHV-33575] Temporarily removing MHV SM authentication to facilitate development

mmoyer-va committed:
[#MHV-33798] Added some routes for SM app

Commit:
Successfully wrote and ran passing simple test that renders header without errors

Commit:
erased comment

Commit:
Merge branch 'MHV-33578-first-unit-test' into mhv-sm-main

KolbyVA committed:
[#MHV-33798] First work, just got write access

KolbyVA committed:
[MHV-33798] First version of Reply Form

Commit:
Created menu from desktop view on the va.gov design page and edited to display messages list, still need to add font awesome icon and need to switch the close (x) button on next to myHealth as opposed to next to Messages

mattwrightva committed:
MHV-33554: Started MHV SM compose new message page

mattwrightva committed:
MHV-33554:  Basic mobile compose structure finished

KolbyVA committed:
[MHV-33798] Mobile Reply view done, some components need work, but are shared with other containers

mattwrightva committed:
Merge branch 'MHV-33554-mobile-compose-structure' into mhv-sm-main

KolbyVA committed:
Merge branch 'MHV-33798-reply-page-mobile' into mhv-sm-main

Commit:
Merge branch 'mhv-sm-main' of https://github.com/department-of-veterans-affairs/vets-website into MHV-33796-Mobile-Menu-Navigation

Commit:
Finished styling the menu, just need it to not close when mobile breakpoint is hit

mattwrightva committed:
MHV-33584: Compose structure / layout for desktop complete

mattwrightva committed:
MHV-33584: Tweaks to compose desktop structure/layout

mmoyer-va committed:
Merge branch 'main' into mhv-sm-main

KolbyVA committed:
Desktop view done

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33584-compose-structure-desktop

KolbyVA committed:
[#MHV-33802] Breadcrumbs layout + duplicate error

KolbyVA committed:
Merge branch 'mhv-sm-main' into MHV-33799-reply-page-desktop

mattwrightva committed:
MHV-33584: Unit tests added for Compose container and ComposeForm component

alex-morgun-1 committed:
[#MHV-33546] message detail page - mobile

alex-morgun-1 committed:
Merge branch 'MHV-33546-message-detail-mobile' into mhv-sm-main

Commit:
Menu now docks to the top right, removed icon component and instead used vas class name of va-btn-close-icon which adds the icon instead

mattwrightva committed:
MHV-33584: Unit tests fixed for ComposeForm and written for MessageCategoryInput component

Commit:
Removed navigation from routes

mattwrightva committed:
MHV-33584: Tweaks to compose views and created compose scss file

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33584-compose-structure-desktop

mmoyer-va committed:
[#MHV-33533] Added conditionally displayed auth/unauth landing pages

mmoyer-va committed:
Merge branch 'mhv-sm-main' into MHV-33533-landing-page-layout

Commit:
Issue with unit test not passing was scss import, ended up putting all of my scss into the main css file and removing the scss import, now the unit test passes

Commit:
changed the unit test to check for nav bar header

Commit:
seperated navigation styling from from secure-messageing.scss and imported it instead as a seperate sass syle sheet

mattwrightva committed:
MHV-33594: Search messages mobile view complete plus other fixes

mattwrightva committed:
MHV-33594: Deleted lowercase breadcrumbs component

mattwrightva committed:
MHV-33594: Search messages mobile and desktop view complete

mattwrightva committed:
MHV-33801: Search results mobile and desktop layout complete

Commit:
Cleaned up ordering of elements, class naming and styling

Commit:
cleaned up the style sheet

Commit:
Merge branch 'mhv-sm-main' of https://github.com/department-of-veterans-affairs/vets-website into mhv-33796-mobile-menu-navigation

Commit:
removed style that hid the x button

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33801-search-results-mobile-view

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33594-search-messages-mobile-view-structure

alex-morgun-1 committed:
[#MHV-33785] message details desktop

alex-morgun-1 committed:
Merge branch 'mhv-sm-main' of https://github.com/department-of-veterans-affairs/vets-website into mhv-33785-message-detail-desktop

alex-morgun-1 committed:
Merge branch 'mhv-sm-main' of https://github.com/department-of-veterans-affairs/vets-website into mhv-33785-message-detail-desktop

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33594-search-messages-mobile-view-structure

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33801-search-results-mobile-view

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33594-search-messages-mobile-view-structure

mattwrightva committed:
MHV-33801: Renamed breadcrumbs to pascal case

mmoyer-va committed:
[#MHV-33533] Built out the MHV SM landing page

mmoyer-va committed:
Merge remote-tracking branch 'origin' into MHV-33533-landing-page-layout

mmoyer-va committed:
Merge branch 'mhv-sm-main' into MHV-33533-landing-page-layout

# Conflicts:
#	src/applications/mhv/secure-messaging/components/breadcrumbs.jsx

mmoyer-va committed:
[#MHV-33533] Fixed breadcrumbs import

mmoyer-va committed:
[#MHV-33533] Fixed styling on the message-sort control

mmoyer-va committed:
[#MHV-33533] Organized components

Commit:
[#MHV-33797] was able to make non messages disapear on mobile view, added navigation to app component, still need to create burger menu, closing functionality on mobile view, and reorder nav so that main messages view is to the right

mmoyer-va committed:
[#MHV-33533] Updated inbox styling to better match designs

KolbyVA committed:
[#MHV-35261] added unit tests for reply page

KolbyVA committed:
Merge branch 'mhv-sm-main' into MHV-35261-reply-page-unit-test

mmoyer-va committed:
Merge branch 'mhv-sm-main' into MHV-33533-landing-page-layout

Commit:
[MHV-33797] Finished Styling component

Commit:
[MHV-33797] giving the navigation hamburger button a button type changed its apeparance on hover, so i took it off

Commit:
[MHV-33797] Placed navigation on app.jsx so that its always placed either on the left of main content or above main content. Used the flex style to align it on both mobile and desktop views

Commit:
Merge branch 'mhv-sm-main' of https://github.com/department-of-veterans-affairs/vets-website into mhv-33797-desktop-menu-navigation

# Conflicts:
#	src/applications/mhv/secure-messaging/components/Navigation.jsx
#	src/applications/mhv/secure-messaging/containers/App.jsx

mmoyer-va committed:
Merge branch 'main' into mhv-sm-main

mmoyer-va committed:
Removed an outdated unit test

Commit:
finished with changes requested by ucd in validation meeting

Commit:
Merge branch 'mhv-sm-main' of https://github.com/department-of-veterans-affairs/vets-website into mhv-33797-desktop-menu-navigation

Commit:
changed unit test description

alex-morgun-1 committed:
[#MHV-35381] view older messages

functionality to expand/collapse older messages + styling to match designs

alex-morgun-1 committed:
[#MHV-35381] minor styling tweaks

alex-morgun-1 committed:
Merge branch 'main' into mhv-35381-older-messages-styling

alexandec commented on src/applications/mhv/secure-messaging/components/OlderMessages.jsx:
```diff
@@ -1,22 +1,77 @@
+
+  return (
+    <div className="older-messages">
+      <div
```
Consider adding `aria-expanded`/`aria-controls` here

oleksii-morgun-va replied:
added as well. Thank you for pointing that out. New to aria and accessibility attributes, pardon any inconsistencies

alexandec commented on src/applications/mhv/secure-messaging/components/OlderMessages.jsx:
```diff
@@ -1,22 +1,77 @@
+      {isExpanded && (
+        <div>
+          <div className="vads-l-row vads-u-justify-content--flex-end">
+            <span
```
See above

oleksii-morgun-va replied:
done

alexandec commented on src/applications/mhv/secure-messaging/components/OlderMessages.jsx:
```diff
@@ -3,20 +1,77 @@
+          Older messages in this conversation
+          {isExpanded ? (
+            <i className="fas fa-angle-up fa-lg" />
+          ) : (
```
Consider using `aria-hidden` here

oleksii-morgun-va replied:
done

alexandec commented on src/applications/mhv/secure-messaging/components/OlderMessages.jsx:
```diff
@@ -1,22 +1,77 @@
+        <span className="vads-u-font-weight--bold">
+          Older messages in this conversation
+          {isExpanded ? (
+            <i className="fas fa-angle-up fa-lg" />
```
Consider using `aria-hidden` here

oleksii-morgun-va replied:
applied. Thank you!

alexandec commented on src/applications/mhv/secure-messaging/sass/secure-messaging.scss:
```diff
@@ -145,6 +152,33 @@
+    transition-property: color, background-color, border-color;
+  }
+  i {
+    margin: 0.5rem;
```
Consider using a design system class instead. Same thing for these other selectors

oleksii-morgun-va replied:
refactored where applicable. some designs we know for sure will change, so will keep that in mind for future edits

alexandec commented on src/applications/mhv/secure-messaging/components/OlderMessages.jsx:
```diff
@@ -3,20 +1,77 @@
+              }}
+            >
+              Expand All Messages
+              {isExpandedAll ? (
```
See above

alexandec commented on src/applications/mhv/secure-messaging/components/OlderMessages.jsx:
```diff
@@ -3,20 +1,77 @@
+              Expand All Messages
+              {isExpandedAll ? (
+                <i className="fas fa-angle-up" />
+              ) : (
```
See above

oleksii-morgun-va replied:
done

alex-morgun-1 committed:
[#MHV-35381] adding aria-hidden on icons

alex-morgun-1 committed:
[#MHV-35381] adding accessibility attributes

aria-expanded and aria-controls on buttons that expand older messages and expand message body on all messages

alex-morgun-1 committed:
[#MHV-35381] replacing css styling with formation classes
